### DISCUSSION TITLE
System Keyboard (TouchScreenKeyboard) Options - specifically numeric keypad

### DISCUSSION THREAD

**Discussion by mevsie** [microsoft/MixedRealityToolkit-Unity#9558] [Q&A] [answered]
We've been trying to bring up a numeric keypad for entering decimal values. To do this we've tried to make use of the [TouchScreenKeyboard.Open](https://docs.unity3d.com/ScriptReference/TouchScreenKeyboard.Open.html) method. 

I'm aware the [documentation]( https://docs.microsoft.com/en-gb/windows/mixed-reality/develop/unity/keyboard-input-in-unity) states that this interaction requires XAML mode. This does work in XAML however the user experience is jarring - dropping in and out of 3D and with XAML placeholder windows. With D3D mode, the keyboard comes up, but it ignores the parameters and doesn't show the prompt contents.

Reading the Slack pages it seems the suggestion is to roll your own or use the Experimental [NonNativeKeyboard](https://localjoost.github.io/migrating-to-mrtk2-using-non-native/) - however that doesn't seem to have a numeric pad mode.

I'm seeking some advice on possible way's forward

1) Roll our own keypad 
2) Try again with XAML and see if there are some settings to make it less horrible
3) Wait for the next MRTK/HL update where this may be available
4) Use another technique/approach that we don't know about.

Anyhow I appreciate your time reading through this question and would appreciate any input

Cheers

**Comment by MaxWang-MS** [comment, marked as answer]
There are several known Unity-side keyboard bugs that we are tracking. All known bugs will be fixed in Unity 2020.3.2 when it comes out (we are not sure which 2019.4 release will contain the fixes). But even on Unity 2020.3.2 we are not sure if the numeric keyboard is implemented by Unity, so maybe you can wait until 2020.3.2 is out and then give it a try. If there are no bugs and the numeric keyboard is implemented then you are good to go, if not you can file a bug report with Unity and maybe implement your own keyboard.
